File: src/main/java/com/github/thorben/webnd/domain/character/DetailTypeRepository.java
package com.github.thorben.webnd.domain.character;

import com.github.thorben.webnd.domain.games.GameType;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface DetailTypeRepository extends CrudRepository<DetailType, Integer> {

	List<DetailType> findAllByGameType(GameType gameType);

}

File: src/main/kotlin/com/github/thorben/webnd/domain/character/DetailTypeRepository.kt
package com.github.thorben.webnd.domain.character

import com.github.thorben.webnd.domain.games.GameType
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface DetailTypeRepository : CrudRepository<DetailType, Int> {
	fun findAllByGameType(gameType: GameType): List<DetailType>
}